backend/src/main/scala/com/softwaremill/adopttapir/starter/content/ContentModel.scala

```diff
@@ -1,11 +1,34 @@
 package com.softwaremill.adopttapir.starter.content
 
-sealed trait Node {
+import cats.syntax.all.*
+import io.circe.{Encoder, Decoder}
+import io.circe.*
+import io.circe.parser.*
+import io.circe.syntax.*
+import io.circe.generic.semiauto.*
+import sttp.tapir.Schema
+import io.circe.*
+
+sealed trait Node:
   def name: String
-}
 
-case class File(name: String, content: String) extends Node
-case class Directory(name: String, content: List[Node]) extends Node {
+/**
+  * Here we diverged from using default deriving approach for both
+  * * Schema[Node] due to https://tapir.softwaremill.com/en/latest/endpoint/schemas.html#derivation-for-recursive-types-in-scala3
+  * * Encoder[Node] due to limitations of circe 0.14 for Scala 3 wrt customizing serializers
+  */
+object Node:
+  implicit def schema: Schema[Node] = Schema.derived[Node]
+
+  given Encoder[Node] = Encoder.instance {
+    case file @ File(_, _)     => file.asJson.mapObject(_.add("type", "file".asJson))
+    case dir @ Directory(_, _) => dir.asJson.mapObject(_.add("type", "directory".asJson))
+  }
+
+  given Decoder[Node] = Decoder[File].or(Decoder[Directory].widen)
+
+case class File(name: String, content: String) extends Node derives Codec.AsObject, Schema
```

Comment:
here we add `derives ... Schema`, but e.g. `Error_OUT` doesn't have this. Shouldn't it? Maybe we have somewhere `import sttp.tapir.generic.auto._` which would auto-derive schemas when it shouldn't

Reply:
We are not using `import sttp.tapir.generic.auto._`. There is though that `TapirSchemas` trait that brings up the Schema givens inside `Http` class.

Reply:
Well `trait TapirSchemas extends SchemaDerivation` which is auto-derivation. So I think we should remove that `... extends SchemaDerivation` and see if things compile.

Reply:
I have removed the whole `TapirSchema` trait and added `derives ... Schema` in few more places.